### ISSUE TITLE
Remove unused OOP wrappers for algorithms

### ISSUE THREAD

**Issue by daniel3735928559** [nextml/NEXT#134]
Many of the algorithms are wrapped in classes when they could simply be modules.  For example, rather than having a directory structure of 

```
algs/RandomSamplingLinearLeastSquares/__init__.py
algs/RandomSamplingLinearLeastSquares/RandomSamplingLinearLeastSquares.py
```

with `RandomSamplingLinearLeastSquares.py` defining a class, just have 

`algs/RandomSamplingLinearLeastSquares.py`

containing the functions defined by the algorithm, wrapped in a class (or possibly not, as needed).

**Comment by kgjamieson** [comment]
Some algorithms require helper utility files, e.g. PoolBasedTriplets.
That's why we threw them in a folder instead of keeping them in one file
that could get very messy

**Comment by daniel3735928559** [comment]
Yeah--those are instances where the extra folder structure is used.  As far as I understand, Python does not distinguish between the existing structure and the one suggested above, so if we simplify everything to use the suggested one when possible, leaving open the option "you can create a whole folder with redundantly named files and an empty `__init__.py` file if you want extra separate utils/whatever" but avoiding this mess when we don't want utils or anything.

**Comment by stsievert** [comment]
This is addressed by #163
